projects/packages/sync/src/class-lock.php

```diff
@@ -42,18 +42,37 @@ class Lock {
 	 * @return boolean True if succeeded, false otherwise.
 	 */
 	public function attempt( $name, $expiry = self::LOCK_TRANSIENT_EXPIRY ) {
-		$name        = self::LOCK_PREFIX . $name;
-		$locked_time = get_option( $name );
+
+		global $wpdb;
+		$name = self::LOCK_PREFIX . $name;
+
+		// Options API is explicitly bypassed. This is because with high concurrency request to the
+		// site multiple processes would pass checks and send concurrent data to WP.
+
+		$row         = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", $name ) );
```

Comment:
Random thought. I'd use `get_var` and check for `empty()` below.

Reply:
no longer applies.